torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -5841,15 +5844,83 @@ def test_broadcast_object_list(self):
             single_obj_list = [objects[0]]
             if self.rank != src_rank:
                 self.assertNotEqual(single_obj_list[0], COLLECTIVES_OBJECT_TEST_LIST[0])
-            dist.broadcast_object_list(single_obj_list, src=0)
+            dist.broadcast_object_list(single_obj_list, src=src_rank)
             self.assertEqual(single_obj_list[0], COLLECTIVES_OBJECT_TEST_LIST[0])
 
             # Multiple input objects test
             if self.rank != src_rank:
                 self.assertNotEqual(objects, COLLECTIVES_OBJECT_TEST_LIST)
-            dist.broadcast_object_list(objects, src=0)
+            dist.broadcast_object_list(objects, src=src_rank)
             self.assertEqual(objects, COLLECTIVES_OBJECT_TEST_LIST)
 
+        @require_backend({"nccl", "gloo"})
+        @require_n_gpus_for_nccl_backend(
+            int(os.environ["WORLD_SIZE"]), os.environ["BACKEND"]
```

Comment:
curious, does this only enforce number of GPUs for NCCL, but not gloo?

Reply:
Yes, the `require_n_gpus_for_nccl_backend()` only enforces the number of GPUs for NCCL. My current understanding is that this is the only GPU requirement that we need for this test.

The `map_location` as GPU part of the test is only run if `torch.cuda.is_available()`, in which case the `map_location` is set to `torch.device("cuda:0")`. This does assume a given nomenclature, namely that if `torch.cuda.is_available() == True`, then `cuda:0` must be a valid GPU. Though, I believe this assumption is valid.

Otherwise, GPUs are only needed if the backend is NCCL for the broadcast. Otherwise, if the backend is not NCCL, then the broadcast uses CPU as the device.